.github/workflows/benchmark.yml

```diff
@@ -122,6 +130,65 @@ jobs:
         run: |
             make benchmarks-basic-run \
                 EXTRA_BENCHMARK_FLAGS="--binary_path ${BINARY_DIR}/baseline/ --output_path ${BASELINE_OUTPUT_PATH}"
+      
+      - name: "Compare initial results"
+        id: compare
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+            ./velox/scripts/benchmark-runner.py compare \
+                --baseline_path ${BASELINE_OUTPUT_PATH} \
+                --contender_path ${CONTENDER_OUTPUT_PATH} \
+                --rerun_json_output "benchmark-results/rerun_json_output_0.json" \
+                --do_not_fail
+
+      - name: "Rerun Benchmarks"
+        if: ${{ github.event_name == 'pull_request'}}    
+        working-directory: 'velox'
+        run: |
+          for i in 1 2 3 4 5; do
+            CURRENT_JSON_RERUN="${RESULTS_ROOT}/rerun_json_output_$((${i} - 1)).json"
+            NEXT_JSON_RERUN="${RESULTS_ROOT}/rerun_json_output_${i}.json"
+  
+            if [ ! -s "${CURRENT_JSON_RERUN}" ]; then
+              echo "::notice::Rerun iteration ${i} found empty file. Finalizing."
+              break
+            fi
+
+            echo "::group::Rerun iteration: ${i}"
+            make benchmarks-basic-run \
+                EXTRA_BENCHMARK_FLAGS="--binary_path ${BINARY_DIR}/baseline/ --output_path ${BASELINE_OUTPUT_PATH}/rerun-${i}/ --rerun_json_input ${CURRENT_JSON_RERUN}"
+
+            make benchmarks-basic-run \
+                EXTRA_BENCHMARK_FLAGS="--binary_path ${BINARY_DIR}/contender/ --output_path ${CONTENDER_OUTPUT_PATH}/rerun-${i}/ --rerun_json_input ${CURRENT_JSON_RERUN}"
+
+            ./scripts/benchmark-runner.py compare \
+                --baseline_path ${BASELINE_OUTPUT_PATH}/rerun-${i}/ \
+                --contender_path ${CONTENDER_OUTPUT_PATH}/rerun-${i}/ \
+                --rerun_json_output ${NEXT_JSON_RERUN} \
+                --do_not_fail
+          echo "::endgroup::"
+          done
+          
+
+          echo "::group::Compare final results"
+          ./scripts/benchmark-runner.py compare \
+                --baseline_path ${BASELINE_OUTPUT_PATH} \
+                --contender_path ${CONTENDER_OUTPUT_PATH} \
+                --recursive \
+                --do_not_fail
+          echo "::endgroup::"
+
+          echo "::group::Copy results from last rerun into upload dir"
```

Comment:
For now, can we actually not put the rerun data into the upload dir? The way it's written, we're associating all the reruns with the same `run_id` in Conbench, which might break a few assumptions we have in the server. This was a problem with the implementation on CircleCI, too.

So for now I think we can just upload the initial run's data, and when we implement https://github.com/conbench/conbench/issues/635 correctly, we can come back and revisit what that looks like here. Do you agree @jonkeane ?

Reply:
> This was a problem with the implementation on CircleCI, too.

I was about to say that I just copied over the way it was done in CCI (upload on rerun). Happy to remove/comment out this step for now.

Reply:
Agreed that not sending data is ok if we're worried it'll violate assumptions. Ideally we _would_ send the re-run data so we have a record of it. But that might be best done when we formally have a framework fro doing this like Austin points out